Report 25.09.19, Philipp & Johanna

Part 1: Review of the paper "Impact of genetic variation on three dimensional structure and
function of proteins"


The biological question
The research group was interested in the effect of SNV's (Single Nucleotide Variations) on their respective protein,
regarding the properties of:
1. Activity
2. Aggregation
3. Stability
4. Binding/Dissociation
5. Assembly
6. Rearrangement

Further they also tried to determine if it is possible to infer patterns regarding the 3D-Structure at the site of the mutation
(e.g. are SNV's more often found on the surface or on the inside of proteins?)


Method
To answer these questions data about 374 SNV's were used. These SNV's were chosen by comparing datasets from three different
databases, PDB, UniProt and dbSNP under application of a set of rules to exclude certain SNV's.
After choosing the 234 SNV's of interest, information about position (surface vs. inside), structure at site (loops vs alpha-helix
vs. beta-strand) and effect on the protein was inferred for each each SNV using literature and databases.
Then the research group developed their own software tools to secure the quality of the gathered data.

After having verified the validity of the data it was analysed to answer the originally posed questions. Additionally
the SNV frequency in the population was also determined.


What significant scientific contribution does the paper make?
The significance of the paper lies in the fact that it brings together data and information on SNV's from a wide
range of different databases and peer-reviewed literature. This could potentially be used to automatically determine
the effects (structural and functional), e.g. via computer models.
However the authors of the paper themselves state that this is not yet possibly, partly due to a lack of better tools
to do so.










Part 2: Paper 2
The biological question
They used a comprehensive assay (MSK-IMPACT) in which they compared tumour and normal sequence data from a unique cohort
of more than 10,000 patients. With this information they identified somatic mutations, novel noncoding alterations, and
mutational signatures that were shared by common and rare tumour types.


Method
They used the MSK-IMPACT clinical workflow to generate their data. In this workflow they sampled tumour tissue to find the
changes in the DNA and blood to gain the normal DNA of a patient. After sequencing the reads were paired through a
bioinformatics pipeline to find multiple classes of genomic rearrangements. The gained data were checked manually for
quality and accuracy in a genomic variant database (MPath). Genomic variations were then uploaded to cBioPortal for data
mining and interpretation.


What significant scientific contribution does the paper make?
They showed that it’s possible to sequence tumours and match them to the corresponding ‘neutral’ blood samples
in a large-scale project. With the automatized computational pipelines they created a comprehensive cancer panel in a
time that permits clinical interpretation and utilization of results. In this project they generated a huge dataset of
mutations and genomic rearrangements in 10,945 tumours from 10,335 patients.









Part 3: Superoxide dismutase 2 (MnSOD or SOD2)
Summary of information on PDB:
When searching for "manganese superoxide dismutase" 95 results are found, 30 of which for homo sapiens.
The mutation we are interested in (1VAR) was analysed using X-Ray diffraction. For the wildtype we found two
possible candidates, 2ADQ and 5VF9. We chose 5VF9 as our reference protein because it is the more recent entry (17.05.2017).


For most recent entry with publication record:
- What is the structure determination method? What is the Uniprot ID?
For our most recent native protein, the above mentioned 5VF9, the method of structure determination was X-Ray diffraction as well.

- How many chains does it have?
According to PDB SOD2 is a Homotetramer consisting of a combination of two different chains, A-chain and B-chain.
We had trouble interpreting the information here, because as far as we understand the term homotetramer, SOD2 would
have to be a protein consisting of 4 identical chains.

- What is the sequence length?
The sequence length is 199.

- Which ligands (if any) are present?
The listed ligands are potassium-cations, Mn2+ Ions and phosphate ions.

- Examine the 3D structure using 3D view. Color by chain. Play with all of the options and prepare a structure figure
we have examined the structure, but cannot upload any pictures on GitHub, therefore this part is missing.
