---
title: 'BIO392 - Day 06: Genome file formats and manipulations in terminal'
layout: default
author:
  - "@imallona"
excerpt_separator: <!--more-->
www_link:
categories:
  - days
tags:
  - material
  - "2020"
---

## {{ page.title }}

Izaskun Mallona (email: izaskun.mallona at sib.swiss).

### Morning (on site)

We will have a lecture and run a set of exercises (on site).

* Overview of the standard genomics data formats
   - FASTA
   - FASTQ
   - SAM
   - BED
   - GFF
   - VCF
* Basic file processing for bioinformatics
   - wc, grep, awk
* Exercises


### Afternoon (online)

Literature/resources reading to consolidate the lectures from the morning (online work).

* [SAM v1 format specification](https://samtools.github.io/hts-specs/SAMv1.pdf)
* [BEDtools paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2832824/)
* [0-start, 1-start, open, closed: how do we count](http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/)
* [GFF3 format](https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md)
* [VCF format](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3137218/)

<!--more-->

### Slides

* [Genomic formats slides](https://github.com/compbiozurich/UZH-BIO392/blob/master/course-material/2020/imallona/2_genomics.pdf)

### Exercises

Exercises 1-28 cover FASTA, FASTQ, SAM, BED, GTF and VCF.

* [Exercises](https://github.com/compbiozurich/UZH-BIO392/blob/master/course-material/2020/imallona/3_exercises.md)

### Resources

* [BEDtools intersect](https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html)

#### Cheatsheets:

* [Shell](https://files.fosswire.com/2007/08/fwunixref.pdf)
* [awk](https://gist.github.com/Rafe/3102414)
* [sed](https://linuxize.com/post/how-to-use-sed-to-find-and-replace-string-in-files/)

### File formats at a glance

* [All](https://genome.ucsc.edu/FAQ/FAQformat.html)
* [BED](https://genome.ucsc.edu/FAQ/FAQformat.html#format1)
* [Wiggle](https://genome.ucsc.edu/goldenPath/help/wiggle.html)
* [BedGraph](https://genome.ucsc.edu/goldenpath/help/bedgraph.html)
* [GFF](https://genome.ucsc.edu/FAQ/FAQformat.html#format3)
* [GTF](https://genome.ucsc.edu/FAQ/FAQformat.html#format4)
* [VCF](https://genome.ucsc.edu/FAQ/FAQformat.html#format10.1)


### Report

Please commit to the BIO392 github a markdown report (due date Sept 26 5pm) describing:

- Which are the advantages of BED/coordinate files as compared to storing just sequences?

- Which QC values are tracked during a normal bioinformatic NGS workflow? (from sequencing to variant calling)?

We'd like to store the following information. You can decide to encode them counting by 0, 1, and closed/open at your convenience (but please specify).
 
All intervals are 1000 nt long. They are contiguous (head to tail). All in the plus strand. The first one starts (we'd like to include the start nucleotide too) in position 1000 of chr2. Intervals A and B have an score of 0, and interval C has a score of 1000.

- How would we store this information in BED3? Are we losing any information?
- And in BED6? Are we losing any information?
- And in BED12? Are we losing any information?
- And in the most compact Wiggle as possible? Are we losing any information?
