# Task 0919


## Estimation of computer storage for the 1000 Genomes project
As the paper mentioned, all 2504 individuals were sequenced using both whole-genome sequencing (mean depth = 7.4X) and targeted exome sequencing (mean depth = 65.7X)
### WGS
The size of a perfect genome is about 715MB according to the slides of the course.

Estimation: 2504 * 715MB * 7.4 = 13 TB 
### WES
Since in humans, protein-coding exons constitute 1.5-1.7% of the human genome ([resource](https://www.ncbi.nlm.nih.gov/pubmed/22688718)). The size of a exome is about 715 * 0.016 = 11.4 MB 

Estimation: 2504 * 11.4MB * 65.7 = 1.9 TB 

## Comparisons of different file formats
For storing called variants, VCF could be used. Because VCF is the standard file format for storing variation data. It is used by large scale variant mapping projects such as IGSR and ExAC. Besides, it is unambiguous, scalable and flexible, allowing extra information to be added to the info field. Many millions of variants can be stored in a single VCF file. 

For archival purposes, BAM could be used. Because the BAM format is more compact and allows fast searching after indexing.

For browser visualisation, BED could be used. Because BED format is often used to store coordinate based information about genes and genomic features.

For just representing sequence information and some comments about the sequence, FASTA could be used. Because it's simple and flexible.
