### Activity 25/09/19 - Lars & Sabrina

# Task 1: Papers

## Mutational landscape of metastatic cancer revealed from prospective clinical sequencing of 10,000 patients

### What is the biological question?

NGS tests allow much higher-throughput clinical approaches in sequencing related to molecular pathology than previously possible. However, the methods are complex and many of the mutations are not very well studied yet, so these kinds of approaches have not been implemented extensively. The goal of the study was to provide a large-scale data set as a clinical resource for identifying novel biomarkers in order to inform prognosis and predict response and resistance to therapy.

### What is the method?

Broad, prospective genomic sequencing was done on >10,000 patients with advanced cancer and available pathological and clinical annotations, by using the MSK-IMPACT assay – an advanced NGS panel capable of detecting many cancer-associated mutations. Tumor sequence data for each subject was then matched and compared with normal sequence data (from unmutated blood cells of the same subject), and used to identify clinically relevant somatic mutations, novel noncoding alterations and mutational signatures shared by both common and rare tumor types.

### What significant scientific contribution does the paper make?

The results of the study provide insights into the biological processes that govern cancer progression and metastasis, as well as help to promote the enrollment of patients onto clinical trials following NGS-based tumor profiling. They also help improve clinical treatment decisions – and subsequently the outcomes for patients. To encourage further investigation, all results are made publicly accessible.


## Impact of genetic variation on three dimensional structure and function of proteins

### What is the biological question?

Linking human phenotypes to genotypic variations at the nucleotide level and the resultant changes in 3D protein structure is vitally important in genomics, as knowledge of the 3D structure of a protein (or a DNA or RNA construct) helps in understanding its function and role in disease. The Protein Data Bank is a very useful resource for this, archiving >126,000 experimentally determined atomic level 3D structures and providing a variety of useful tools. In some cases, different genetic variants of a protein have been determined, revealing the atomic-level structural differences caused by nucleotide variation. The goal is to present a systematic and qualitative analysis of such cases in order to better understand the relationship between point mutations and experimentally observed consequences, and subsequently the heritability of complex diseases.

### What is the method?

A benchmark dataset of 374 protein structure-relevant SNVs was constructed, through a multi-step manual filtering procedure, from the 2596 total structures in the PDB for which 3D atomic coordinates are available and non-synonymous SNVs could be mapped. This set was then analyzed in regards to the mutations' effects on the structure, function, stability, and binding properties of the proteins. Systematic reviews of available literature and searches in several databases were done for the various SNVs in order to classify each into one or more functional categories (Activity, Aggregation, Stability, Binding, Assembly, Rearrangement).

### What significant scientific contribution does the paper make?

The data collected provides important insights into possible structural and functional changes in proteins – though it is only a qualitative description of possible changes. The results show that the range of possible SNV effects at the protein level are significantly greater than previously assumed using existing software prediction methods. A main reason for this found in the study is that point mutations which do not cause disease because they have only a relatively minor effect – at least on their own – can still cause significant changes in a protein which may perturb normal human physiology in some fashion.


# Task 2: Protein

## 5-Amino Laevulinic Acid Dehydratase

5-Amino Laevulinic Acid Dehydratase (alternatively, Delta-Aminolevulinic Acid Dehydratase; ALAD) catalyzes an early step in tetrapyrrole biosynthesis. ALAD Porphyria results from a number of known mutations of the ALAD gene, including F12L, a Phe->Leu SNV missense mutation. The mutation changes the preferred protein assembly from octamer to hexamer and increases the optimal pH for enzyme activity from pH 7 to pH 9. Therefore, it affects both enzymatic activity and protein assembly.

### How many entries are there for this protein?
23

### Which method(s) were used in characterization of the structure?
X-ray Diffraction (all 23)

### For the most recent entry with a publication record:
### What is the structure determination method?
X-ray Diffraction

### What is the Uniprot ID?
5MHB

### How many chains does it have?
Homooctomer - A8 (1 unique chain, 8 total chains)

### What is the sequence length?
324 per chain, 2592 total

### Which ligands (if any) are present?
3 unique ligands:\
ZN (Zinc Ion) – 3 per chain, 24 total\
GOL (glycerol) – 3 per chain, 24 total\
PBG (3-[5-(aminomethyl)-4-(carboxymethyl)-1h-pyrrol-3-yl]propanoic acid) – 1 per chain, 8 total
